ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNodeImpl.java

```diff
@@ -164,6 +177,10 @@ public final boolean hasUnseenUpdates() {
   public void dirty() {
     if (!isVirtual()) {
       mYogaNode.dirty();
+    } else if (getParent() != null) {
```

Comment:
(This is great and should be enough for now, but... ) I believe this will not work in all cases.
Let's imagine that the deeply nested view inside interleaving Text and View hierarchy was changed. This view is not virtual, so it will not call `getParent().dirty()`. As a result of calling `dirty`, Yoga will propagate the dirty flag to the closes root node, and that's it; the propagation will stop on the nearest inline View.

To fix this we have to additionally implement `onDirty` handler in Yoga that will dirty the semi-virtual (virtual outside, non-virtual inside) anchor View node (the inline view node) that will propagate this dirty call to ancestors until the first non-virtual one calls to Yoga again.

Reply:
You were right. I was able to craft cases where inline images and views did not get rerendered. You can see them in the `ChangeImageSize`, `ChangeViewSize`, and `ChangeInnerViewSize` functions in `TextInlineView.js` in [my branch](https://github.com/rigdern/react-native/commit/ae11179a492283a7ab7bb7ef2b49e399eb8717c8#diff-f8c2073073da018fc6d2ed9a4e71f441). I'll be sending my branch as a separate PR so we can have these inline view examples in RNTester.

I added a TODO comment in `ReactShadowNodeImpl.java` to note this bug.